src/Server/Configuration/DidChangeConfigurationProvider.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Disposables;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Primitives;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace OmniSharp.Extensions.LanguageServer.Server.Configuration
+{
+    class DidChangeConfigurationProvider : BaseWorkspaceConfigurationProvider, IDidChangeConfigurationHandler,
+        IOnStarted, ILanguageServerConfiguration
+    {
+        private readonly ILanguageServer _server;
+        private DidChangeConfigurationCapability _capability;
+        private readonly ConfigurationRoot _configuration;
+
+        private readonly ConcurrentDictionary<Uri, DisposableConfiguration> _openScopes =
+            new ConcurrentDictionary<Uri, DisposableConfiguration>();
+
+        public DidChangeConfigurationProvider(ILanguageServer server, Action<IConfigurationBuilder> configurationBuilderAction)
+        {
+            _server = server;
+            var builder = new ConfigurationBuilder()
+                .Add(new DidChangeConfigurationSource(this));
+            configurationBuilderAction(builder);
+            _configuration = builder.Build() as ConfigurationRoot;
+        }
+
+        public async Task<Unit> Handle(DidChangeConfigurationParams request, CancellationToken cancellationToken)
+        {
+            // null means we need to re-read the configuration
+            // https://github.com/Microsoft/vscode-languageserver-node/issues/380
+            if (request.Settings == null || request.Settings.Type == JTokenType.Null)
+            {
+                await GetWorkspaceConfiguration();
+                return Unit.Value;
+            }
+
+            ParseClientConfiguration(request.Settings);
```

Comment:
Do you have examples of a client that gives back xml as configuration (just my knowledge/testing)?  I'll look at adding the service this weekend.

Reply:
So by default VS stores its configurations in XML format however the LSP client for VS doesn't currently support interacting with those XML based configurations. Therefore, for the time being they brought over JSON configurations to act as a replacement of the XML based configurations but all of the new JSON configs do not translate to XML and do not auto-map to the XML configs either. Therefore VS is trying to figure out how it wants its configuration story to roll out so it can work with new language server.

tl;dr; I don't know of any currently; however, it's definitely possible it'll come down in the future.